plugins/rule_engines/irods_rule_language/src/functions.cpp

```diff
@@ -1970,6 +1970,33 @@ Res *smsi_remoteExec( Node** paramsr, int, Node* node, ruleExecInfo_t* rei, int,
 #endif
 }
 
+/**
+ * \fn writeLine(msParam_t *where,  msParam_t *inString, ruleExecInfo_t *rei)
+ *
+ * \brief  Writes a message to either the server log, stdout, stderr, or a data object.
+ *
+ * \module core
+ *
+ * \usage See clients/icommands/test/rules/
+ *
+ * \param[in] where - a msParam of type STR_MS_T which is either "stdout", "stderr", "rodsLog", or a full path to a data object.
+ * \param[in] inString - a msParam of type STR_MS_T which is a string to be logged. 
+ * \param[in,out] rei - The RuleExecInfo structure that is automatically
+ *    handled by the rule engine. The user does not include rei as a
+ *    parameter in the rule invocation.
+ *
+ * \DolVarDependence none
+ * \DolVarModified none
+ * \iCatAttrDependence none
+ * \iCatAttrModified none
+ * \sideeffect none
+ *
+ * \return integer
+ * \retval 0 upon success
+ * \pre N/A
+ * \post N/A
+ * \sa N/A
+**/
 Res *smsi_writeLine( Node** paramsr, int, Node*, ruleExecInfo_t* rei, int, Env* env, rError_t*, Region* r ) {
```

Comment:
We can't use the `smsi_` functions for documenting system/builtin MSIs due to their parameters.

We have to trick doxygen into producing the appropriate function signature. The following demonstrates how you can achieve that. The key is the preprocessor check. That macro is only defined when doxygen documentation is being produced.
https://github.com/irods/irods/blob/06ed23911a6fc3f6f4a35d81cf917bfa3e366a1a/plugins/microservices/src/json_parse.cpp#L105-L148

Reply:
I think I did it how it is expected.